src/main/java/org/ihtsdo/snowowl/authoring/single/api/service/dao/UiStateResourceService.java

```diff
@@ -40,19 +40,27 @@ public final void write(final String path,
 	}
 
 	@Override
-	public final String read(final String path) throws IOException {
+	public final String read(final String path) throws AmazonS3Exception {
 		if (path == null) {
 			throw new PathNotProvidedException("Panel path is null while trying to read the resource stream.");
 		}
-		return IOUtils.toString(resourceManager.readResourceStream(path), StandardCharsets.UTF_8);
+		try {
+			return IOUtils.toString(resourceManager.readResourceStream(path), StandardCharsets.UTF_8);
+		} catch (IOException e) {
+			throw new AmazonS3Exception(e.getMessage(), e);
+		}
 	}
 
 	@Override
-	public final String read(final File file) throws IOException {
+	public final String read(final File file) throws AmazonS3Exception, FileNotFoundException {
 		if (file == null) {
 			throw new FileNotFoundException("File is null while trying to read the resource stream.");
 		}
-		return IOUtils.toString(resourceManager.readResourceStream(file.getPath()), StandardCharsets.UTF_8);
+		try {
+			return IOUtils.toString(resourceManager.readResourceStream(file.getPath()), StandardCharsets.UTF_8);
+		} catch (IOException e) {
+			throw new AmazonS3Exception(e.getMessage(), e);
+		}
```

Comment:
Here looks a bit strange to wrap the IOException to AmazonS3Exception. Especially the original AmazonS3Exception is wrapped into the IOException by the ResourceManager.

Reply:
@CoderMChu  @Tyler-Cooksey Please see my solution here: https://github.com/IHTSDO/authoring-services/commit/85b7d6f566914007219150963761f5dbbb9008b9